From caaeacf381bcf0d8bd7506c0bda7d3bcc1aa5622 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Sun, 15 Nov 2020 16:20:31 +0200
Subject: [PATCH] [SW-0]: goya2: resolve build errors

while type casting from u64 to void * need to take care of
different processors(ARC, ARM) compilation

Change-Id: I6107e48c85c886f1c4752922471337f6c58ecc07
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/ppboot/ppboot.c b/ppboot/ppboot.c
index b044a80..7a040db 100644
--- a/ppboot/ppboot.c
+++ b/ppboot/ppboot.c
@@ -24,7 +24,7 @@
 {
 	u8 hash[SHA256_SUM_LEN];
 
-	sha256_csum((void *)desc->loadaddr, desc->img_size, hash);
+	sha256_csum((void *)(uintptr_t)desc->loadaddr, desc->img_size, hash);
 
 	return !memcmp(hash, desc->chksum, ARRAY_SIZE(hash));
 }
